tools/eclipse-shadow-generator/advanced_modes.cu

```diff
@@ -0,0 +1,535 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//                               This file is part of CosmoScout VR                               //
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// SPDX-FileCopyrightText: German Aerospace Center (DLR) <cosmoscout@dlr.de>
+// SPDX-License-Identifier: MIT
+
+#include "advanced_modes.cuh"
+
+#include "atmosphere_rendering.cuh"
+#include "common.hpp"
+#include "gpuErrCheck.hpp"
+#include "math.cuh"
+#include "tiff_utils.hpp"
+
+#include <cstdint>
+#include <iostream>
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+namespace {
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+enum class Mode { eShadow, eLimbLuminance, ePlanetView, eAtmoView };
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// Tonemapping code and color space conversions.
+// http://filmicworlds.com/blog/filmic-tonemapping-operators/
+
+__device__ glm::vec3 uncharted2Tonemap(glm::vec3 c) {
+  const float A = 0.15;
+  const float B = 0.50;
+  const float C = 0.10;
+  const float D = 0.20;
+  const float E = 0.02;
+  const float F = 0.30;
+  return ((c * (A * c + C * B) + D * E) / (c * (A * c + B) + D * F)) - E / F;
```

Comment:
It is hard to read with having an upper and lower case c in the same equation. Maybe use UVWXYZ or write color instead of c.

Reply:
Done.